### PR TITLE
Update quantum tapes serialization and Python tests

### PR DESCRIPTION
**Pull request by maliasadi**
**Context:**
Refactoring operations in PennyLane will deprecation a few operations' features including `matrix`, `eigvals` and `control_wires`, and this breaks serialization of quantum tapes and Python tests in Lightning. This PR addresses all `DeprecationWarning` caused by PR [#2023](https://github.com/PennyLaneAI/pennylane/pull/2023) and also fixes `ComplexWarning` and `UserWarning` issues with tests.

### CODE REVIEW

File: pennylane_lightning/lightning_qubit.py
Comment (mlxd) [inline pennylane_lightning/lightning_qubit.py:199]: :+1:

File: tests/test_gates.py
Comment (mlxd) [inline tests/test_gates.py:66]: I always worry about this assumed control vs target wire combination (makes life tricky for certain backend libs).
Not a comment for changing here though, just in general.

File: tests/test_serialize.py
Comment (mlxd) [inline tests/test_serialize.py:299]: Are these separated out now into different tests?

**Review by mlxd — COMMENTED** [review]
Nice work @maliasadi 
Just a quick question on the parameterized tests, but happy to approve once that is answered.

File: pennylane_lightning/_serialize.py
Comment (josh146) [inline pennylane_lightning/_serialize.py:116]: @maliasadi no action needed for now, but this is a temporary name that still might change before the next release - I'll update you once I know the final decision!

File: tests/test_gates.py
Comment (maliasadi) [reply tests/test_gates.py:66]: Me too. FYI, this is the refactored PL code for initializing `wires` and `control_wires`, 
```
        if control_wires is None:
            if len(wires) > 1:
                control_wires = Wires(wires[:-1])
                wires = Wires(wires[-1])
            else:
                raise ValueError(
                    "MultiControlledX: wrong number of wires. 1 wire given. Need at least 2."
                )
        else:
            wires = Wires(wires)
            control_wires = Wires(control_wires)

            warnings.warn(
                "The control_wires keyword will be removed soon. Use wires = (control_wires,target_wire). See the documentation for more information.",
                category=UserWarning,
            )
```

Comment (josh146) [reply tests/test_gates.py:66]: I think we plan on changing this! (cc @albi3ro)

File: tests/test_serialize.py
Comment (maliasadi) [reply tests/test_serialize.py:299]: Yes. There are test cases (`test_integration_c64` and `test_integration_c128`) for each one. I had to remove this to prevent `ComplexWarning` due to some casting complex values to real issues.

File: pennylane_lightning/_serialize.py
Comment (mlxd) [reply pennylane_lightning/_serialize.py:116]: Should we go ahead with merging this, or is it best to await on the updated name?

Comment (maliasadi) [reply pennylane_lightning/_serialize.py:116]: I think that's better to wait until the op refactoring of the coming release gets settled.

Comment (maliasadi) [reply pennylane_lightning/_serialize.py:116]: There are definitely going to be more changes and so we can address all within this PR.

Comment (maliasadi) [reply pennylane_lightning/_serialize.py:116]: Hi @josh146. Would it be any further updates on ops requiring changes/fixes in Lightning soon? I want to know when is suitable for this PR to be merged.

Comment (josh146) [reply pennylane_lightning/_serialize.py:116]: Hey @maliasadi, you can update this now to use `qml.matrix(o_)` 🙂 then it should be good to merge!

File: tests/test_gates.py
Comment (albi3ro) [reply tests/test_gates.py:66]: The call signature for `MultiControlledX` was changed recently to bring it in line with other gates like `CNOT`, `Toffoli`, `CRX`, etc.

The new signature will remain, but it will instead be a constructor for an operation of type `Controlled`.

File: pennylane_lightning/_serialize.py
Comment (mlxd) [inline pennylane_lightning/_serialize.py:30]: Should we use the same structure of 
```
from pennylane import matrix
```
or something like
```
from pennylane import matrix as qml_mat
```

to avoid pulling in the full library?

Comment (mlxd) [reply pennylane_lightning/_serialize.py:30]: Though it is probably fine.

**Review by mlxd — APPROVED** [review]
Thanks @maliasadi 
I left one comment, but happy to approve either option :)

File: pennylane_lightning/_serialize.py
Comment (maliasadi) [reply pennylane_lightning/_serialize.py:30]: Ah! I fix it now.

Comment (maliasadi) [reply pennylane_lightning/_serialize.py]: Hi @mlxd, would you check if this is completely fixed in both `_serialized` and `lightning_qubit`?

Comment (maliasadi) [reply pennylane_lightning/_serialize.py]: Ok. Tests are failed! Because the `qml.matrix` is new and not supported with current versions of PL. So, we should wait till the next release of PL.

**Review by mlxd — APPROVED** [review]
Nice one @maliasadi 
Happy to merge!

**Pull request merged**
